### DISCUSSION TITLE
What method to retrieve results is faster?

### DISCUSSION THREAD

**Discussion by montyclt** [ravendb/ravendb#13904] [Q&A] [answered]
Hi, there are multiple methods to retrieve single results (`Single()`, `Firsts()`) and multiple results (`ToArray()`, `ToList()`...)

What of these methods has better performance?

I know that MongoDB using `First` is faster than `Single` because `First` limit to 1 and `Single` limit to 2, using the second result to check that only exists one in the database. I know that EF using `ToList` is faster than `ToArray` because need to determine the array size and copied.

Whats about RavenDB?

**Comment by ppekrol** [comment, marked as answer]
Hi,

due to the nature of `Single` it takes 2 elements and `First` takes 1 - same as in MongoDB. 

When it comes to `ToList` and `ToArray` the only difference would be the allocation of those and the performance of adding an element inside .NET itself. From RavenDB this is the only difference in a code path between those two, one method allocates an array and adds elements via `array[index] = element`, second one allocates a list and uses `list.Add(element)` method.

Hope that it helps.
